src/coco/repository/util/repositoryUtil.ts

```diff
@@ -0,0 +1,86 @@
+import RepositoryType from "../model/RepositoryType";
+import Repository from "../model/Repository";
+import { execSync } from "child_process";
+import Logger from "../../../log/Logger";
+
+const GET_GIT_REPOSITORY_COMMAND = "git config --local --get remote.origin.url";
+const GIT_REPO_URL_SUFFIX = ".git";
+const DOMAIN_GITHUB = "github.com";
+const DOMAIN_GITLAB = "gitlab.com";
+const DOMAIN_BITBUCKET = "bitbucket.org";
+const CUSTOM_DOMAIN_REG_EXP_GITHUB = /github.(.+).com/; // TODO
+const SUFFIX_HTTPS = "https://";
+
+function getRepositoryForType(rootFolderPath: string, type: RepositoryType): Repository {
+    let fullUrl: string | undefined;
+    try {
+        fullUrl = execSync(GET_GIT_REPOSITORY_COMMAND, {
+            cwd: rootFolderPath
+        }).toString().trim();
+    } catch (error) {
+        Logger.error(`Getting git repository url failed`, error);
+    }
+
+    if (!fullUrl) {
+        return getDefaultRepository();
+    }
+
+    if (fullUrl.endsWith(GIT_REPO_URL_SUFFIX)) {
+        fullUrl = fullUrl.substring(0, fullUrl.length - GIT_REPO_URL_SUFFIX.length);
+    }
+
+    const defaultUrl = getDefaultUrlForType(type);
+    const defaultUrlIndex = fullUrl.indexOf(defaultUrl);
+    if (defaultUrlIndex >= 0) {
+        const name = fullUrl.substring(defaultUrlIndex + defaultUrl.length + 1); // +1 is for the divider after domain
+        return {
+            repository: name
+        };
+    }
+
+    const customDomainRegExp = getCustomDomainRegExpForType(type);
+    const customUrl = customDomainRegExp?.exec(fullUrl)?.[0];
+    if (customUrl) {
+        const customUrlIndex = fullUrl.indexOf(customUrl);
+        const name = fullUrl.substring(customUrlIndex + customUrl.length + 1); // +1 is for the divider after domain
+        const fullCustomUrl = SUFFIX_HTTPS + customUrl;
+        return {
+            repository: name,
+            url: fullCustomUrl
+        };
+    }
+
+    return getDefaultRepository();
+}
+
+function getDefaultRepository(): Repository {
+    return {
+        repository: ""
+    };
+}
+
+function getDefaultUrlForType(type: RepositoryType): string {
+    switch (type) {
+        case RepositoryType.Github:
+            return DOMAIN_GITHUB;
+        case RepositoryType.Gitlab:
+            return DOMAIN_GITLAB;
+        case RepositoryType.Bitbucket:
+            return DOMAIN_BITBUCKET;
+        default:
+            throw new Error(`Unhandled repository type: ${type}`);
```

Comment:
Future suggestion/improvement: we can still handle all types of repos, but for the unrecognized ones we show the logo of git itself in the clients.

Reply:
The only option of repository on config is github, gitlab and bitbucket. This error is for developers only. That code should never be reached.

Reply:
I know, I'm suggesting that we can support any git repo in the future if we want. It's just a custom url but we know that it's a git repo, so we can show git logo.

Reply:
We do not show a logo anywhere. What do you mean here?

Reply:
> I know, I'm just suggesting that we can support any git repo in the future if we want.

👍